main.py

```diff
@@ -18,38 +25,41 @@ def parse_data(data):
 
 
 def get_houses_list(data):
-    return data['web_widgets']['post_list']
+    return data["web_widgets"]["post_list"]
 
 
 def extract_each_house(house):
-    data = house['data']
+    data = house["data"]
 
     return {
-        'title': data['title'],
-        'description': data['description'],
-        'district': data['district'],
-        'token': data['token'],
+        "title": data["title"],
+        "description": f'{data["top_description_text"]}  {data["middle_description_text"]}',
+        "district": data["action"]["payload"]["web_info"]["district_persian"],
+        "hasImage": data["image_count"] > 0,
+        "token": data["token"],
     }
 
 
 def send_telegram_message(house):
-    url = 'https://api.telegram.org/bot' + BOT_TOKEN + '/sendMessage'
-    text = f"<b>{house['title']}</b>"+"\n"
-    text += f"<i>{house['district']}</i>"+"\n"
-    text += f"{house['description']}"+"\n\n"
+    url = "https://api.telegram.org/bot" + BOT_TOKEN + "/sendMessage"
+    text = f"<b>{house['title']}</b>" + "\n"
+    text += f"<i>{house['district']}</i>" + "\n"
+    text += f"{house['description']}" + "\n"
+    if house["hasImage"]:
+        text += f"<i>تصویر : </i>" + "✅" + "\n\n"
+    else:
+        text += f"<i>تصویر : </i>" + "❌" + "\n\n"
     text += f"https://divar.ir/v/a/{house['token']}"
-
-    body = {
-        'chat_id': BOT_CHATID,
-        'parse_mode': 'HTML',
-        'text': text
-    }
-
-    requests.post(url, data=body)
+    body = {"chat_id": BOT_CHATID, "parse_mode": "HTML", "text": text}
+    result = requests.post(url, data=body)
+    if not result.status_code == 200:
+        time.sleep(5)
+        send_telegram_message(house)
 
 
 def load_tokens():
-    with open("tokens.json", "r") as content:
+    my_path = str(pathlib.Path(__file__).parent.resolve()) + "//"
+    with open(my_path + "tokens.json", "r") as content:
```

Comment:
I don't get the reason, can you explain a little please?

Reply:
when this application runs with cronjob it couldn't locate the token files 
this approach uses full path to solve the problem

Reply:
In this case, please use `os.path.join`, and avoid using `/` or `\` directly.
Something like this:
```py
token_path = os.path.join(
    os.path.dirname(os.path.realpath(__file__)),
    "tokens.json"
)
with open(token_path, "r") as content:
```